internal/cmd/oohelperd/internal/nwcth/internal.go

```diff
@@ -0,0 +1,51 @@
+package nwcth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/ooni/probe-cli/v3/internal/iox"
+	"github.com/ooni/probe-cli/v3/internal/version"
+)
+
+const maxAcceptableBody = 1 << 24
+
+// Handler implements the Web Connectivity test helper HTTP API.
+type NWCTHHandler struct{}
+
+func (h NWCTHHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	w.Header().Add("Server", fmt.Sprintf(
+		"oohelperd/%s ooniprobe-engine/%s", version.Version, version.Version,
+	))
+	if req.Method != "POST" {
+		w.WriteHeader(400)
+		return
+	}
+	if req.Header.Get("content-type") != "application/json" {
+		w.WriteHeader(400)
+		return
+	}
```

Comment:
On second though, I think this is too strict. What if the client sets `application/json; charset=UTF-8`? I think we should be liberal in what we accept and just assume the body is JSON and parse it as such.

Reply:
(If this is also a problem of the current test helper, could you also fix it there?)

Reply:
I removed the check from both `internal.go` files.